.test-infra/jenkins/Infrastructure.groovy

```diff
@@ -0,0 +1,109 @@
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import CommonJobProperties as common
+import CommonTestProperties.SDK
+
+class Infrastructure {
+
+  static void prepareSDKHarness(def context, SDK sdk, String repositoryRoot, String dockerTag) {
+    context.steps {
+      String sdkName = sdk.name().toLowerCase()
+      String image = "${repositoryRoot}/${sdkName}"
+      String imageTag = "${image}:${dockerTag}"
+
+      shell("echo \"Building SDK harness for ${sdkName} SDK.\"")
+      gradle {
+        rootBuildScriptDir(common.checkoutDir)
+        common.setGradleSwitches(delegate)
+        tasks(":beam-sdks-${sdkName}-container:docker")
+        switches("-Pdocker-repository-root=${repositoryRoot}")
+        switches("-Pdocker-tag=${dockerTag}")
+      }
+      shell("echo \" Tagging Harness' image\"...")
+      shell("docker tag ${image} ${imageTag}")
+      shell("echo \" Configure docker credentials\"...")
+      shell("docker-credential-gcr configure-docker")
+      shell("echo \" Pushing Harness' image\"...")
+      shell("docker push ${imageTag}")
+    }
+  }
+
+  static void prepareFlinkJobServer(def context, String repositoryRoot, String dockerTag) {
+    context.steps {
+      String image = "${repositoryRoot}/flink-job-server"
+      String imageTag = "${image}:${dockerTag}"
+
+      shell('echo "Building Flink job Server"')
+
+      gradle {
+        rootBuildScriptDir(common.checkoutDir)
+        common.setGradleSwitches(delegate)
+        tasks(":beam-runners-flink_2.11-job-server-container:docker")
+        switches("-Pdocker-repository-root=${repositoryRoot}")
+        switches("-Pdocker-tag=${dockerTag}")
+      }
+
+      shell("echo \" Tagging Flink Job Server's image\"...")
+      shell("docker tag ${image} ${imageTag}")
+      shell("echo \" Configure docker credentials\"...")
+      shell("docker-credential-gcr configure-docker")
+      shell("echo \" Pushing Flink Job Server's image\"...")
+      shell("docker push ${imageTag}")
+    }
+  }
+
+  static void setupFlinkCluster(def context, String clusterNamePrefix, Integer workerCount, String imagesToPull, String jobServerImage) {
+    String gcsBucket = 'gs://beam-flink-cluster'
+    String clusterName = getClusterName(clusterNamePrefix)
+    String artifactsDir="${gcsBucket}/${clusterName}"
+
+    context.steps {
+      environmentVariables {
+        env("GCLOUD_ZONE", "us-central1-a")
+        env("CLUSTER_NAME", clusterName)
+        env("GCS_BUCKET", gcsBucket)
+        env("FLINK_DOWNLOAD_URL", 'http://archive.apache.org/dist/flink/flink-1.5.6/flink-1.5.6-bin-hadoop28-scala_2.11.tgz')
+        env("FLINK_NUM_WORKERS", workerCount)
```

Comment:
Should we use a later version of flink? Does Beam support 1.8? Otherwise 1.7?

Reply:
Good catch - I should have used at least 1.7 here. It's the highest version I was able to use here. I know that there's 1.8 support in beam too but this caused problems - since flink 1.8 flink is packaged without hadoop28 and I cannot setup a cluster on dataproc due to that. 

I created JIRA addressing problems with 1.8: https://issues.apache.org/jira/browse/BEAM-7440

Reply:
Support for Flink 1.8 has been added recently (will be part of 2.13.0). Yes, you will have to include the Hadoop jar in the classpath, or add a Gradle dependency on Hadoop for your tests.

Reply:
Shall we parameterize it for later flink updates and multiversion tests.

Reply:
Parametrization is indeed a low hanging fruit - I'll fix this